code/modules/mob/living/simple_animal/hostile/commanded/commanded.dm

```diff
@@ -17,16 +18,73 @@
 	if(!short_name)
 		short_name = name
 
+/mob/living/simple_animal/hostile/commanded/AltClick(mob/user)
+	if(user != master)
+		to_chat(user, SPAN_WARNING("You can't command \the [src], you aren't their master!"))
+		return
+	var/list/options = list()
+	for(var/command in known_commands)
+		var/command_icon_state = "crossed_out"
+		switch(command)
+			if("stay")
+				command_icon_state = "down_arrow"
+			if("stop")
+				command_icon_state = "crossed_out"
+			if("attack")
+				command_icon_state = "targeting_glyph"
+			if("follow")
+				command_icon_state = "right_arrow"
+		command = capitalize(command)
+		var/image/radial_button = image(icon = 'icons/radial/basic.dmi', icon_state = command_icon_state)
+		options[command] = radial_button
+	var/choice = show_radial_menu(user, src, options, radius = 42, tooltips = TRUE)
+	if(choice)
+		choice = lowertext(choice)
+		var/suffix = ""
+		switch(choice)
+			if("stay")
+				suffix = "here"
+			if("attack")
+				var/list/attack_list = list()
+				for(var/mob/M in view(user.client.view, user))
+					if(M == user || M == src)
+						continue
+					var/radial_button = get_mob_head_mutable_appearance(M)
+					attack_list[M.name] = radial_button
+				suffix = show_radial_menu(user, src, attack_list, radius = 42, tooltips = TRUE)
+				if(!suffix)
+					choice = null
+			if("follow")
+				var/list/attack_list = list()
+				for(var/mob/M in view(user.client.view, user))
+					if(M == src)
+						continue
```

Comment:
The way you build this list here and in "attack" are basically the same, I'm pretty sure you can unify the two in some way